src/components/rangeselector/get_update_object.js

```diff
@@ -42,13 +39,13 @@ function getXRange(axisLayout, buttonLayout) {
 
     switch(buttonLayout.stepmode) {
         case 'backward':
-            range0 = Lib.ms2DateTime(+d3.time[step].utc.offset(base, -count));
+            range0 = axisLayout.l2r(+d3.time[step].utc.offset(base, -count));
             break;
 
         case 'todate':
             var base2 = d3.time[step].utc.offset(base, -count);
 
-            range0 = Lib.ms2DateTime(+d3.time[step].utc.ceil(base2));
+            range0 = axisLayout.l2r(+d3.time[step].utc.ceil(base2));
```

Comment:
>  I propose to leave this as an open item for now, because it's going to have all sorts of edge cases to sort out 

I agree :+1: . But let's make the range selector supplyDefaults doesn't allow `axis.calendar !== 'gregorian'` with `axis.rangeselector.buttons[i].stepmode = 'todate'`